Inferences/Induction.cpp

```diff
@@ -674,4 +721,13 @@ bool InductionClauseIterator::notDone(Literal* lit, Term* term)
   return true;
 }
 
+Term* InductionClauseIterator::getPlaceholderForTerm(Term* t) {
+  CALL("InductionClauseIterator::getPlaceholderForTerm");
+
+  unsigned termType = env.signature->getFunction(t->functor())->fnType()->result();
+  unsigned placeholderConstNumber = env.signature->addFreshFunction(0, "placeholder");
```

Comment:
I'm a little concerned about creating a large number of fresh placeholder constants. I think it's not possible to have a single placeholder but is there a way to reuse. My understanding is that these objects are just used locally within the induction process? Could we have a finite number and keep track of how many we have used so far in this particular induction call?

Reply:
You're right. The placeholder constant is actually never used in a formula in the saturation set, so we can create just one per each type. I changed this in the code.